### ISSUE TITLE
Enhancement: [strict-boolean-expressions] Allow `!!` for casting to a Boolean

### ISSUE THREAD

**Issue by tylerlaprade** [typescript-eslint/typescript-eslint#7444]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/strict-boolean-expressions/

### Description

I like using `!!` for casting to a Boolean because it's more concise than `Boolean()`. My _optimal_ solution would be for there to be an option to configure the suggested fix to always use `!!`, but I'd be perfectly happy with simply making `!!` no longer a violation.

### Fail

```typescript
function f(obj: unknown, foo: string) {
    return obj && foo;
}
```

### Pass

```typescript
function f(obj: unknown, foo: string) {
    return !!obj && foo;
    // OR
    return Boolean(obj) && foo;
}
```

### Additional Info

This core rule treats both as valid, equivalent ways to cast to a Boolean:
https://eslint.org/docs/latest/rules/no-extra-boolean-cast

**Comment by tylerlaprade** [comment]
>To be honest even Boolean() should be banned by the rule - it's kind of a hole that it isn't checked right now.

It's this inconsistency that I'd like to address. Not only does the rule allow `Boolean()`, but it currently suggests this as an automatic fix in the IDE, which is a tacit endorsement of the practice. Given that both `!!` and `Boolean()` are considered equivalent by an ESLint core rule, they should both be treated the same. If that means this rule also rejects `Boolean()`, that would make sense to me.
